### DISCUSSION TITLE
Can I have some details on use_adaptive_sync?

### DISCUSSION THREAD

**Discussion by semi-hollow** [HewlettPackard/swarm-learning#87] [Q&A]
swarm_callback = SwarmCallback(
             sync_interval = <batch interval as integer>,
             min_peers = <minimum peers count as an integer>,
             use_adaptive_sync = False <boolean>,
             val_data = <either a (x_val, y_val) tuple or a generator>,
             val_batch_size = <batch size of val_data as an integer>,
             checkin_model_on_train_end = 'snapshot' <how to checkin model once its local training is over>,
             node_weightage = 1 <node weightage as an interger between 0 to 100>,
             ml_platform= <ML platform -'TF'/'KERAS'/'PYTORCH' as a string>, 
             model_name='swarm_model' <model name as a string>
        )
        
I'd like to know more details about the use_adaptive_sync = False <boolean>, how it get the sync interval automatically?  based on what parameters or some other factors?  What would happen If I have set a specific sync interval number and also set the adaptive_sync to True?

**Comment by RadhakrishnaJ** [comment]
No product issue here, moving this to discussion forum. 

'use_adaptive_sync' specifies whether the adaptive sync interval feature should be used for automatically generating the the sync interval. This feature is turned off by default.

when 'use_adaptive_sync' is not set, swarm learning uses fixed sync interval through out learning. 

When use_adaptive_sync is enabled, swarm generates next sync interval based on loss computation. Sync interval mentioned in callback used only for first time merge process, subsequent sync intervals are dynamically computed by swarm based on trend of loss. This feature judges the training progress by monitoring the mean loss. A reduction in the mean loss infers that the training is progressing well. As a response, it increases the sync interval and enables more batches to run before sharing the
learnings. This makes the training run faster. In contrast, when the loss does not improve, Adaptive Sync frequency
reduces the sync frequency and synchronizes the models more frequently.
